src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java

```diff
@@ -111,16 +128,45 @@ boolean isAutoStart()
 		return ( (ContainerCraftConfirm) this.inventorySlots ).isAutoStart();
 	}
 
+	private void reinitalize()
+	{
+		this.buttonList.clear();
+		this.initGui();
+	}
+
 	@Override
 	public void initGui()
 	{
+		this.maxRows = this.getMaxRows();
+
+		final int magicNumber = ITEMSTACK_TOP_OFFSET + CPU_TOP_OFFSET;
+		final int extraSpace = this.height - magicNumber;
+
+		this.rows = (int) Math.floor( extraSpace / SECTION_HEIGHT );
+		if( this.rows > this.maxRows )
+		{
+			this.rows = this.maxRows;
+		}
+
+		if( this.rows < 5 )
+		{
+			this.rows = 5;
+		}
```

Comment:
The whole sanitation should probably already happening in `getRows()` or `getMaxRows()`.